facebook/OpenBIC
# fby3.5: rf: Support GPIO interrupt

Summary:
- Support monitor system DC state.
- Support monitor MB reset.

Known issue:
- HW: ASIC voltage is 1.8V, but it's too low to let GPIO high.

Test plan:
1. Check system DC flag will change.
- MB power on
[BMC console]
root@bmc-oob:~# power-util slot2 on
Powering fru 2 to ON state...
root@bmc-oob:~# power-util slot2 status
Power status for fru 2 : ON

[RF BIC console]
uart:~$ [set_MB_DC_status] gpio number(32) status(1)
[control_power_on_sequence] power on success
[set_DC_status] gpio number(33) status(1)

- MB power off
[BMC console]
root@bmc-oob:~# power-util slot2 off
Powering fru 2 to OFF state...
root@bmc-oob:~# power-util slot2 status
Power status for fru 2 : OFF

[RF BIC console]
uart:~$ [set_MB_DC_status] gpio number(32) status(0)
[set_DC_status] gpio number(33) status(0)
[control_power_off_sequence] power off success

2. Check ASIC_PERST0_N will change when host reset.
[BMC console]
root@bmc-oob:~# power-util slot2 reset

[RF BIC console]
uart:~$ platform gpio get 1
[1  ] ASIC_PERST0_N                      : OD  | output(I) | 0(0)
uart:~$ platform gpio get 1
[1  ] ASIC_PERST0_N                      : OD  | output(O) | 0(0)
uart:~$ platform gpio get 1
[1  ] ASIC_PERST0_N                      : OD  | output(I) | 0(0)
